### PR TITLE
New: Collections View

### LINKED ISSUE #5476: Include "Add Collection" option when adding a new movie.
**Is your feature request related to a problem? Please describe.**
This isn't really a solution to a _problem_ but it would make life easier...

When adding a new movie, I often want to add all the movies in the collection. If it's a series like _Mad Max_ a search for "Mad Max" will likely show all the movies in the top page of results. However, if I search for _London Has Fallen_, I'm not going to get the rest of the _...Has Fallen_ collection showing up, and I certainly won't get all the movies in the series if I search for _Die Hard_. (Whether I _should_ or not is a totally different discussion, not for this forum. 🤣)

**Describe the solution you'd like**
On the `Add New Movie` dialogue, there should be a check box (enabled/visible only if the movie is IDd as a part of a collection) to `Add list - TMDB collection`. When checked, this would _either_:
* Add the collection using the `add collection` logic that already exists and using all the parameters defined for the individual movie, (preferable) or
* Go directly to the `Add Collection` dialogue, prepopulated with the parameters just used for adding the individual movie. (Less favorable from a UX perspective, but probably easier from a code perspective.)

**Describe alternatives you've considered**
The alternative is to leave it as is and force people to go to the individual movie to enable the Collection option.

**Additional context**
Having the `Add Collection` available as part of the `Add Movie` process would improve discoverability of the `Collections` feature added in v3.

[AB#786](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/786)

### LINKED ISSUE #6861: Better Collection Support
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe

It would be good if there was a page setup to view collections of movies in your list, that aren't monitored. Also movie collection suggestions, and an option to automatically monitor collections of added movies.

### Describe the solution you'd like

Add an extra page under Discover, for Collections and also add a checkbox option to automatically monitor collections of added / imported movies.

### Describe alternatives you've considered

going through my entire library looking for any collection check box that is not monitored :|

### Anything else?

I think it is described well enough.

[AB#2248](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/2248)

### PR DESCRIPTION
**Pull request by Qstick**
#### Database Migration
YES - 208

- Add Collections Table
- Migrate collections from movies
- Migrate collection monitored status from Collection typed lists
- Remove collection typed lists
- Add CollectionId Column to Movies Table
- Migrate Column ShouldMonitor to Monitor on Import Lists for new monitoring types
- Remove Collection column from Movie (Now LazyLoaded)

#### Description
This PR makes a few QOL enhancements for collections:

1. Dedicated UI view for collections allowing the user to easily monitor/adjust collections and movies in collection from one place.
    - Options to control what is shown, poster size
    - Custom filters based on title and monitored status
    - Mass update of collections and movies in them
    - Status of movies currently in library
    - Monitor toggles for collections and movies
    - New horizontal slider control for movies that exceed screen width

2. API for collections, `/collection` allowing one to retrieve all collections, edit collection defaults, and add missing movies from collections
3. Rework monitoring options for movie from a bool to enum with 3 values. `movieOnly`, `movieAndCollection`, `none`. This allows collections to be automatically monitored on adding of a movie meaning users and 3rd party programs can ensure monitored collections without additional actions. 

#### Screenshot (if UI related)
image

#### Todos
- [x] Unit Tests
- [x] Migration Tests
- [x] Housekeeper for Collections without movies
- [x] Adding of movies from Collections view
- [ ] Mobile tweaks
- [ ] Code cleanup and final review
- [ ] Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
- [ ] [Wiki Updates](https://wiki.servarr.com)

#### Issues Fixed or Closed by this PR

* Fixes #6861 
* Fixes #5476

### CODE REVIEW

File: src/NzbDrone.Core/Datastore/Migration/207_collections.cs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    public class collections : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            Create.TableForModel("Collections")
+                .WithColumn("TmdbId").AsInt32().Unique()
+                .WithColumn("QualityProfileId").AsInt32()
+                .WithColumn("RootFolderPath").AsInt32()
+                .WithColumn("MinimumAvailability").AsInt32()
+                .WithColumn("SearchOnAdd").AsInt32()
```
Comment (RobinDadswell) [inline src/NzbDrone.Core/Datastore/Migration/207_collections.cs:24]: Should this be a boolean?

File: src/NzbDrone.Core/Datastore/Migration/207_collections.cs [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+            Delete.Column("Collection").FromTable("Movies");
+        }
+
+        private void MigrateCollections(IDbConnection conn, IDbTransaction tran)
+        {
+            var rootPaths = new List<string>();
+            using (var getRootFolders = conn.CreateCommand())
+            {
+                getRootFolders.Transaction = tran;
+                getRootFolders.CommandText = @"SELECT Path FROM RootFolders";
```
Comment (RobinDadswell) [inline src/NzbDrone.Core/Datastore/Migration/207_collections.cs:56]: Postgres support the command please

File: src/NzbDrone.Core/Datastore/Migration/207_collections.cs [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+                        string path = definitionsReader.GetString(0);
+                        rootPaths.Add(path);
+                    }
+                }
+            }
+
+            using (var cmd = conn.CreateCommand())
+            {
+                cmd.Transaction = tran;
+                cmd.CommandText = "SELECT Collection, ProfileId, MinimumAvailability, Path FROM Movies WHERE Collection IS NOT NULL GROUP BY Collection";
```
Comment (RobinDadswell) [inline src/NzbDrone.Core/Datastore/Migration/207_collections.cs:71]: More escapes for postgres support

**Review by RobinDadswell — CHANGES_REQUESTED** [review]
Some postgres fixes so I don't need to do it if this merges first. 

Also does the "missing root folder" health check need extending to include when a collection includes a root folder but the root folder doesn't exist in Radarr

File: src/NzbDrone.Core/Datastore/Migration/207_collections.cs (hunk 1 above)
Comment (Qstick) [reply src/NzbDrone.Core/Datastore/Migration/207_collections.cs:24]: Yes

File: src/NzbDrone.Core/Datastore/Migration/207_collections.cs [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+                .WithColumn("CleanTitle").AsString()
+                .WithColumn("Overview").AsString().Nullable()
+                .WithColumn("Images").AsString().WithDefaultValue("[]")
+                .WithColumn("Monitored").AsBoolean().WithDefaultValue(false)
+                .WithColumn("LastInfoSync").AsDateTime().Nullable()
+                .WithColumn("Added").AsDateTime().Nullable()
+                .WithColumn("Movies").AsString().WithDefaultValue("[]");
+
+            Alter.Table("Movies").AddColumn("CollectionId").AsInt32().Nullable();
+            Alter.Table("ImportLists").AddColumn("Monitor").AsInt32().Nullable();
```
[the remark this answers is not in the document]
Comment (Qstick) [reply src/NzbDrone.Core/Datastore/Migration/207_collections.cs:36]: No, it's an enum

File: src/NzbDrone.Core/MetadataSource/SkyHook/SkyHookProxy.cs [hunk 5]
```diff
@@ -98,6 +99,35 @@ public HashSet<int> GetChangedMovies(DateTime startTime)
             return new Tuple<Movie, List<Credit>>(movie, credits.ToList());
         }
 
+        public MovieCollection GetCollectionInfo(int tmdbId)
+        {
+            var httpRequest = _radarrMetadata.Create()
+                                             .SetSegment("route", "movie/collection")
+                                             .Resource(tmdbId.ToString())
+                                             .Build();
```
[the remark this answers is not in the document]
Comment (Qstick) [reply src/NzbDrone.Core/MetadataSource/SkyHook/SkyHookProxy.cs:107]: Already is

**Pull request merged**